### PR TITLE
refactor, wallet: Nuke coincontrol circular dependency

### PR DESCRIPTION
**Pull request by hebasto**
This PR gets rid of `wallet/coincontrol` -> `wallet/wallet` -> `wallet/coincontrol` circular dependency.

### CODE REVIEW

**Comment by practicalswift** [comment]
Concept ACK

Very nice :)

**Comment by Sjors** [comment]
ACK 80303fa2b9a8eaaa00d8c3ddf665a5026675508d. Tested on macOS 10.15.1

File: src/wallet/init.cpp [hunk 1]
```diff
@@ -11,6 +11,7 @@
 #include <util/moneystr.h>
 #include <util/system.h>
 #include <util/translation.h>
+#include <wallet/coincontrol.h> // for DEFAULT_AVOIDPARTIALSPENDS
```
Comment (promag) [inline src/wallet/init.cpp]: While here also add `#include <ui_interface.h>` because of `InitError`.

nit, does the comment really matter?

**Review by promag — COMMENTED** [review]
ACK 80303fa2b9a8eaaa00d8c3ddf665a5026675508d.

File: src/wallet/coincontrol.h [hunk 2]
```diff
@@ -6,14 +6,18 @@
...
 #include <policy/fees.h>
 #include <primitives/transaction.h>
-#include <wallet/wallet.h>
+#include <script/standard.h>
 
 const int DEFAULT_MIN_DEPTH = 0;
 const int DEFAULT_MAX_DEPTH = 9999999;
 
+//! Default for -avoidpartialspends
+static constexpr bool DEFAULT_AVOIDPARTIALSPENDS = false;
```
[the remark this answers is not in the document]
Comment (hebasto) [reply src/wallet/coincontrol.h]: > I've tried to move these to a new header...

That was the first attempt too ;)

File: src/wallet/init.cpp (hunk 1 above)
Comment (hebasto) [reply src/wallet/init.cpp]: Done.

**Comment by hebasto** [comment]
@promag @Sjors 

Thank you for your reviews. @promag's comments have been addressed.

Would you mind re-reviewing?

**Comment by hebasto** [comment]
Rebased after #16944 has been merged.

**Comment by Sjors** [comment]
re-ACK 3ed5e6819a50434449d92cb96b9d8d141e8c7d2b

**Review by meshcollider — COMMENTED** [review]
utACK 3ed5e6819a50434449d92cb96b9d8d141e8c7d2b

**Pull request merged**